openapi-cli/src/test/java/io/ballerina/openapi/generators/common/TestConstants.java

```diff
@@ -251,4 +251,40 @@ public class TestConstants {
             "Enabled by default\n" +
             "    boolean validation = true;\n" +
             "|};";
+
+    public static final String CONNECTION_CONFIG_HTTP_VERSION_1_1 = "" +
+            "# Provides a set of configurations for controlling the behaviours when communicating " +
+            "with a remote HTTP endpoint.\n" +
+            "@display {label: \"Connection Config\"}\n" +
+            "public type ConnectionConfig record {|\n" +
+            "    # The HTTP version understood by the client\n" +
+            "    http:HttpVersion httpVersion = http:HTTP_1_1;\n" +
```

Comment:
Will we need to check `ConnectionConfig` record compatibility for http versions 1.0, 1.1, and  2.0 since we have modified `ConnectionConfig` record by adding fields such as  `http:ProxyConfig proxy`, `boolean validation` ?

Reply:
Adding compatible the values for the fields is a responsibility of the connector user. The `ConnectionConfig` record which is a copy of ballerina/http's `[CommonClientConfiguration](https://github.com/ballerina-platform/module-ballerina-http/blob/d3705117c518d0ebaafb9464b93327adae55eb16/ballerina/http_types.bal#L67)` record is same for any httpVersion. httpVersion is a field of that record. 

@AathmanT is there any fields in the `CommonClientConfiguration` that are specific to http1.1 or not compatible with http1 version? Appreciate your feedback.

Reply:
The `CommonClientConfiguration` only has the configurations common to both HTTP/1.1 and HTTP/2. The configurations specific to the HTTP version are separated in `ClientHttp1Settings` and `ClientHttp2Settings`.